services/web/src/components/form-fields/Roles.js

```diff
@@ -22,45 +38,243 @@ export default class Roles extends React.Component {
+    return union(
+      roles.filter((role) => role.scope === scope).map((role) => role.scopeRef)
+    );
+  }
+
+  fetchOrganizations = async () => {
+    const { data } = await request({
+      method: 'POST',
+      path: '/1/organizations/search',
+      body: {},
+    });
+    return data;
+  };
+
+  addOrganizationScope() {
+    const { currentOrganization, scopedOrganizationIds } = this.state;
+    scopedOrganizationIds.push(currentOrganization.id);
+    this.setState({ scopedOrganizationIds });
+  }
+
+  getOptionsForScope(scope) {
+    const { allRoles } = this.state;
+    return Object.keys(allRoles)
+      .map((key) => {
+        return {
+          ...allRoles[key],
+          id: key,
+        };
+      })
+      .filter((role) => role.allowScopes.includes(scope))
+      .map((role) => {
+        return {
+          value: role.id,
+          key: role.id,
+          text: role.name,
+        };
+      });
+  }
+
   render() {
-    const { required = true, label = 'Roles', value, onChange } = this.props;
-    const { error, loading, roles } = this.state;
+    const {
+      error,
+      loading,
+      scopedOrganizationIds,
+      showAddOrganizationModal,
+    } = this.state;
+    const { enableOrganizationScopes } = this.props;
     if (loading) return <Loader />;
     if (error) return <Message error content={error.message} />;
     return (
-      <Form.Dropdown
-        name="roles"
-        label={label}
-        required={required}
-        fluid
-        selection
-        multiple
-        value={value.map((role) => role.role) || []}
-        options={Object.keys(roles).map((role) => {
-          return {
-            value: role,
-            key: role,
-            text: roles[role].name,
-          };
+      <>
+        <Form.Dropdown
+          name="globalRoles"
+          label={
+            <>
+              Global Roles
+              <Popup
+                content="Global scoped roles give a user permissions across all organizations."
+                on="click"
+                trigger={
+                  <Icon
+                    style={{
+                      marginLeft: '5px',
+                      cursor: 'pointer',
+                      color: '#cccccc',
+                    }}
+                    name="help circle"
+                  />
+                }
+              />
+            </>
+          }
+          fluid
+          selection
+          multiple
+          value={this.getGlobalValues()}
+          options={this.getOptionsForScope('global')}
+          onChange={(e, { value }) => this.setGlobalValues(value)}
+        />
+        {scopedOrganizationIds.map((organizationId) => {
+          const name = `organization-${organizationId}`;
+          return (
+            <FetchObject
+              key={name}
+              id={organizationId}
+              endpoint="organizations">
+              {(organization) => {
+                return (
+                  <Form.Dropdown
```

Comment:
FYI `FetchObject` + `Form.Dropdown` is exactly the purpose of `src/components/form-fields/Reference`... may need tweaking but conceptually it's the exact same thing